### DISCUSSION TITLE
Prune Html failed because of CDN

### DISCUSSION THREAD

**Discussion by may7243** [LuXDAmore/nuxt-prune-html#54] [Q&A] [answered]
Hi!

We just apply the module to our website, but the function seems failed because of CloudFront CDN.
We notice that your website (https://www.fallimenti.it/) uses CloudFront CDN as well, and we like to know how did you setup your CDN settings to make it work?

The module is really a life-saver!
Thanks for advance!

**Comment by LuXDAmore** [comment, marked as answer]
We're using nuxt in server-side-mode (so not spa nor generate), our configuration looks like this one:

```js
// nuxt.config.js
export default {
    pruneHtml: {
            hookGeneratePage: false,
            types: [ 'headers-exist' ],
            headersToPrune: [
                {
                    key: 'cloudfront-is-bot-viewer',
                    value: 'true',
                },
                {
                    key: 'cloudfront-is-audit-viewer',
                    value: 'true',
                },
            ],
        },
    }
};
```

Also, our DevOps team is setting this 2 additional headers with Cloudfront:
- `cloudfront-is-bot-viewer`;
- `cloudfront-is-audit-viewer`.

So, as you can see in the `headersToPrune` key, if the value in one of the 2 provided headers is `'true'`, the prune will happen!

---

Some resources about Cloudfront Headers:
- [User-Agent header](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html#request-custom-user-agent-header);
- [Adding the CloudFront HTTP headers](https://docs.amazonaws.cn/en_us/AmazonCloudFront/latest/DeveloperGuide/using-cloudfront-headers.html);
- [AWS CloudFront - forward User-Agent](https://stackoverflow.com/a/55222635/3446499).


> Hope this helps! 🐱

**Reply by may7243** [reply]
We will definitely check this out! Thanks a lot!!!
